pymc/aesaraf.py

```diff
@@ -970,3 +975,26 @@ def compile_pymc(
         **kwargs,
     )
     return aesara_function
+
+
+def constant_fold(xs: Sequence[TensorVariable]) -> Tuple[np.ndarray, ...]:
+    """Use constant folding to get constant values of a graph.
+
+    Parameters
+    ----------
+    xs: Sequence of TensorVariable
+        The variables that are to be constant folded
+
+    Raises
+    ------
+    NotConstantValueError:
+        If any of the variables cannot be successfully constant folded
+    """
+    fg = FunctionGraph(outputs=xs, features=[ShapeFeature()], clone=True)
+
+    folded_xs = rewrite_graph(fg, custom_rewrite=topo_constant_folding).outputs
+
+    if not all(isinstance(folded_x, Constant) for folded_x in folded_xs):
+        raise NotConstantValueError
```

Comment:
What is the implication if we remove this? `folded_xs` would just be a tensor?
I am thinking that instead of raising error, maybe we can make the function being "best effort constant fold".

Reply:
It would be a tensor, but we cloned the graph, so it has nothing to do with the original one (except for shared inputs)

Reply:
LOL i looking at this line and feeling we are being a bit more too aggressive - seems my hunch was justified.

Reply:
You mean the follow up PR? If so, yeah I thought I had already pushed it and you were looking at it. My bad :P